src/Compiler/Compile.hs

```diff
@@ -584,9 +614,14 @@ searchModule flags currentDir name
                          Nothing -> searchPackageIface flags currentDir Nothing name
                          Just iface -> return (Just iface)
 
+getCurrentFileTime :: FilePath ->  (FilePath -> Maybe (BString, FileTime)) -> IO FileTime
+getCurrentFileTime fp maybeContents =
+  case maybeContents fp of
+    Just (_, t) -> return t
+    Nothing -> getFileTimeOrCurrent fp
```

Comment:
This method gets the current file time, giving preference to a `FileTime` reported by the virtual file system.